orca/producer.go

```diff
@@ -100,16 +101,21 @@ type producer struct {
 	// reports a result.
 	backoff func(int) time.Duration
 
-	mu        sync.Mutex
-	intervals map[time.Duration]int    // map from interval time to count of listeners requesting that time
-	listeners map[OOBListener]struct{} // set of registered listeners
+	mu           sync.Mutex
+	intervals    map[time.Duration]int    // map from interval time to count of listeners requesting that time
+	listeners    map[OOBListener]struct{} // set of registered listeners
+	hasIntervals chan struct{}            // created when intervals is empty; closed and nilled when non-empty.
```

Comment:
Can we use a `sync.Cond` instead? It seems like an appropriate use case for a condition variable. I wanted to try it out before asking for it. And it does seem simpler to use a condition variable instead of closing and recreating the channel.